assets/js/modules/analytics/components/common/PropertySelectIncludingGA4.js

```diff
@@ -0,0 +1,127 @@
+	const ga4PropertyID = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getPropertyID() );
+	const uaPropertyID = useSelect( ( select ) => select( MODULES_ANALYTICS ).getPropertyID() );
+	const isLoading = useSelect( ( select ) => {
+		const isLoadingAccounts =	! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getAccounts' );
+		const isLoadingPropertiesGA4 =	! select( MODULES_ANALYTICS_4 ).hasFinishedResolution( 'getProperties', [ accountID ] );
+		const isLoadingProperties =	! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getProperties', [ accountID ] );
+
+		return isLoadingAccounts || isLoadingProperties || isLoadingPropertiesGA4;
+	} );
+
+	const primaryPropertyType = useSelect( ( select ) => select( MODULES_ANALYTICS ).getPrimaryPropertyType() );
+
+	const ga4Dispatch = useDispatch( MODULES_ANALYTICS_4 );
+	const uaDispatch = useDispatch( MODULES_ANALYTICS );
+
+	const propertyID = primaryPropertyType === 'ga4' ? ga4PropertyID : uaPropertyID;
+
+	const onChange = useCallback( ( index, item ) => {
+		const newPropertyID = item.dataset.value;
+		const internalID = item.dataset.internalId; // eslint-disable-line sitekit/acronym-case
+		if ( propertyID !== newPropertyID ) {
+			trackEvent( 'analytics_setup', 'property_change', newPropertyID );
+			if ( !! internalID ) {
+				uaDispatch.selectProperty( newPropertyID, internalID );
+				uaDispatch.setPrimaryPropertyType( 'ua' );
+
+				ga4Dispatch.setPropertyID( '' );
+				ga4Dispatch.setWebDataStreamID( '' );
+				ga4Dispatch.setMeasurementID( '' );
+			} else {
+				ga4Dispatch.selectProperty( newPropertyID );
+				uaDispatch.setPrimaryPropertyType( 'ga4' );
+
+				uaDispatch.setPropertyID( '' );
+				uaDispatch.setInternalWebPropertyID( '' );
+				uaDispatch.setProfileID( '' );
+			}
+		}
+	}, [ propertyID, ga4Dispatch, uaDispatch ] );
+
+	if ( ! isValidAccountID( accountID ) ) {
+		return null;
+	}
+
+	if ( isLoading ) {
+		return <ProgressBar small />;
+	}
+
+	const properties = unmappedProperties.map( ( p ) => ( {
+		...p,
+		id: p._id || p.id,
+		name: p.displayName || p.name,
+	} ) );
+
+	return (
+		<Select
+			className="googlesitekit-analytics__select-property"
+			label={ __( 'Property', 'google-site-kit' ) }
+			value={ propertyID }
+			onEnhancedChange={ onChange }
+			enhanced
+			outlined
+		>
+			{ ( properties || [] )
+				.concat( {
+					id: PROPERTY_CREATE,
+					name: __( 'Set up a new property', 'google-site-kit' ),
+				} )
+				.map( ( { id, name, internalWebPropertyId }, index ) => ( // eslint-disable-line sitekit/acronym-case
+					<Option
+						key={ index }
+						value={ id }
+						data-internal-id={ internalWebPropertyId } // eslint-disable-line sitekit/acronym-case
+					>
+						{ id === PROPERTY_CREATE
+							? name
+							: sprintf(
+								/* translators: 1: Property name. 2: Property ID. */
+								__( '%1$s (%2$s)', 'google-site-kit', 'google-site-kit' ),
```

Comment:
@danielgent the second parameter must be a context, let's better use something like this:

```suggestion
								__( '%1$s (%2$s)', '{property name} ({property id})', 'google-site-kit' ),
```

Reply:
@eugene-manuilov  Now the linting fails with
`118:9  error  Invalid text domain '{property name} ({property id})'  @wordpress/i18n-text-domain`

I'm not sure what this means and (from a quick look) I can't find any examples that have this argument (all the calls to `sprinf` seem to have just two arguments)

Reply:
@danielgent use `_x` instead of `__` and check out https://github.com/WordPress/gutenberg/tree/trunk/packages/i18n when you have some spare time.